src/structs/pdb.rs

```diff
@@ -1051,6 +1051,27 @@ impl<'a> PDB {
         }
         chains
     }
+
+    /// Returns a vector of unique residue names present in the PDB file.
+    ///
+    /// # Arguments
+    ///
+    /// * `self` - A reference to the PDB struct.
+    ///
+    /// # Returns
+    ///
+    /// * `Vec<String>` - A vector of unique residue names.
+    pub fn unique_residue_names(&self) -> Vec<String> {
+        let mut resnames = Vec::new();
+        for residue in self.residues() {
+            let resname = residue.name().unwrap().to_owned();
+            let index = resnames.binary_search(&resname);
+            if index.is_err() {
```

Comment:
Really nice, one style point though: a more Rusty like way of writing this would be:
```rust
if let Some(index) = resnames.binary_search(&resname) {
resnames.insert(index, resname);
}
```

Reply:
thanks! I appreciate that, still getting using to the syntax 😅